File: JULY-2025/22-07-2025/p1695.ts
function maximumUniqueSubarray(nums: number[]): number {
    let seen = new Set<number>();
    let cs = 0, maxSum = 0, start = 0;
    // cs means current sum

    for (let i = 0; i < nums.length; i++) {
        // while loop run till the repeating element get delted
        while (seen.has(nums[i])) {
            seen.delete(nums[start]); // deleting nums[start] not nums[i]
            cs = cs - nums[start];
            start++;
        }

        seen.add(nums[i]);
        cs = cs+ nums[i];
        maxSum = Math.max(maxSum, cs);
    }

    return maxSum;
}

//TC:SC: O(N)

File: JULY-2025/22-07-2025/p1695.java
class Solution {
    public int maximumUniqueSubarray(int[] nums) {
        Set<Integer> set=new HashSet<>();
        int max=0;
        int curr=0;
        int l=0;
        for(int r=0;r<nums.length;r++){
            if(!set.contains(nums[r])){
                set.add(nums[r]);
                curr+=nums[r];
            }
            else{
                max=Math.max(curr,max);
                while(set.contains(nums[r])){
                    curr-=nums[l];
                    set.remove(nums[l]);
                    l++;
                }
                r--;
            }
        }
        max=Math.max(curr,max);
        return max;
    }
}